crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -359,12 +359,46 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
 
         let mut out: UInt32Chunked = ca
             .into_iter()
-            .map(|opt_s| opt_s.map(|s| reg.find_iter(s).count() as u32))
+            .map(|opt_s| {
+                opt_s.map(|s| reg.find_iter(s).count() as u32)
+            })
             .collect();
         out.rename(ca.name());
         Ok(out)
     }
 
+    /// Count all successive non-overlapping regex matches.
+    fn count_match_many(&self, pat: &Utf8Chunked) -> PolarsResult<UInt32Chunked> {
+        let ca = self.as_utf8();
+        polars_ensure!(
+            ca.len() == pat.len(),
+            ComputeError: "pattern's length: {} does not match that of the argument series: {}",
+            pat.len(), ca.len(),
+        );
+
+        let out = ca
+            .into_iter()
+            .zip(pat)
+            .map(|(opt_s, opt_pat)| match (opt_s, opt_pat) {
+                (_, None) | (None, _) => Ok(Some(0)),
+                (Some(s), Some(pat)) => {
+                    let reg = Regex::new(pat)?;
+                    let mut iter = reg.find_iter(s).map(|m| m.as_str()).peekable();
```

Comment:
Why the map or the peeking? Just `.count()` on `reg.find_iter` should work, no?

Reply:
My mistake, I had this at some point but I reverted while finding out how to cast the result of this `map` operation to `UInt32Chunked`. It's fixed now.